components/user/UserProfile.js

```diff
@@ -87,6 +88,24 @@ function UserProfile({ BASE_URL, data }) {
         </div>
       )}
 
+      {data.isStatsPublic && (
+        <BasicCards
+          data={[
+            {
+              name: "Rank",
+              current: data.profilStats?.profile?.rank,
+            },
+            {
+              name: "Total Profile Views",
+              current: data?.profilStats?.profile?.total || 0,
```

Comment:
this can also be updated to be consistent 

```suggestion
              current: data.profilStats?.profile?.total || 0,
```